pkg/ingester/ingester.go

```diff
@@ -547,6 +547,17 @@ func (i *Ingester) stopping(_ error) error {
 	// we need to mark the ingester service as "failed", so Loki will shut down entirely.
 	// The module manager logs the failure `modules.ErrStopProcess` in a special way.
 	if i.terminateOnShutdown && errs.Err() == nil {
+		shutdownMarkerPath := path.Join(i.cfg.ShutdownMarkerPath, shutdownMarkerFilename)
```

Comment:
nitpicking: do you mind moving this check to a separate function?

Reply:
No problems. It's moved.